`Cannot find module 'knex' or its corresponding type declarations.` when using yarn2 PnP with @mikro-orm/migrations

**Describe the bug**

When use yarn2 PnP, importing Migration class from @mikro-orm/migrations will emit the following error: 

```
../../../.yarn/$$virtual/@mikro-orm-migrations-virtual-fbfe7e7b13/0/cache/@mikro-orm-migrations-npm-4.4.3-0e331d383b-f74dac794e.zip/node_modules/@mikro-orm/migrations/Migration.d.ts:17:23 - 
error TS2307: Cannot find module 'knex' or its corresponding type declarations.

17     getKnex(): import("knex")<any, unknown[]>;
```

`@mikro-orm/migrations` should have a dependency on knex, because [this line](https://github.com/mikro-orm/mikro-orm/blob/48d06ac/packages/migrations/src/Migration.ts#L41) is using types from knex, which leads the follwing .d.ts code:

```typescript
getKnex(): import("knex")<any, unknown[]>;
```

**To Reproduce**

Steps to reproduce the behavior:

```bash
mkdir reproduce
cd reproduce
yarn set version berry
yarn init
yarn add typescript @types/node @mikro-orm/core @mikro-orm/migrations
echo "import { Migration } from '@mikro-orm/migrations'" >index.ts
echo "export abstract class SomeMigration extends Migration { }" >> index.ts
yarn tsc index.ts --esModuleInterop
```

**Expected behavior**

No errors are expected.

**Additional context**

N/A

**Workaround**

Add the following config in `.yarnrc.yml` and then run `yarn` again to relink packages:

```yml
packageExtensions:
  "@mikro-orm/migrations@*":
    dependencies:
      knex: "0.21.12"
```

**Versions**

| Dependency | Version |
| - | - |
| node | 12.16.3 |
| typescript | 4.1.3 |
| mikro-orm | 4.4.3 |
| yarn | 2.4.0 |

--- a/packages/migrations/package.json
+++ b/packages/migrations/package.json
@@ -50,6 +50,7 @@
     "@mikro-orm/knex": "^4.4.3",
     "@types/umzug": "2.3.0",
     "fs-extra": "9.1.0",
+    "knex": "0.21.17",
     "umzug": "2.3.0"
   },
   "devDependencies": {

